addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -76,6 +80,8 @@
 
     private static final int HEARTBEAT_INTERVAL = 30;
 
+    private ScheduledExecutorService buttonExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
You should not created your own scheduler here - why not use the "scheduler" flied that is already available to the handler?